services/ux-backend/handlers/featureflags/handler.go

```diff
@@ -0,0 +1,214 @@
+	}
+
+	storageClassExists, err := checkNoobaaStorageClassExists(ctx, client, namespace)
+	if err != nil {
+		return false, fmt.Errorf("failed to check NooBaa StorageClass: %w", err)
+	}
+
+	return storageClassExists, nil
+}
+
+func checkNoobaaCRExists(ctx context.Context, client ctrlclient.Client, namespace string) (bool, error) {
+	klog.Info("Checking NooBaa CR existence in namespace:", namespace)
+
+	noobaaList := &nbv1alpha1.NooBaaList{}
+	if err := client.List(ctx, noobaaList, ctrlclient.InNamespace(namespace), ctrlclient.Limit(1)); err != nil {
+		return false, fmt.Errorf("failed to list NooBaa CRs: %w", err)
+	}
+
+	if len(noobaaList.Items) > 0 {
+		klog.Info("Found NooBaa CRs in namespace:", namespace, "count:", len(noobaaList.Items))
+		return true, nil
+	}
+
+	klog.Info("No NooBaa CRs found in namespace:", namespace)
+	return false, nil
+}
+
+func checkNoobaaStorageClassExists(ctx context.Context, client ctrlclient.Client, namespace string) (bool, error) {
+	klog.Info("Checking for NooBaa StorageClass existence")
+	
+	provisioner := fmt.Sprintf("%s.noobaa.io/obc", namespace)
+	storageClassList := &storagev1.StorageClassList{}
+	if err := client.List(ctx, storageClassList, ctrlclient.MatchingFields{"provisioner": provisioner}, ctrlclient.Limit(1)); err != nil {
+		return false, fmt.Errorf("failed to list StorageClasses by provisioner index: %w", err)
+	}
+	
+	if len(storageClassList.Items) > 0 {
+		klog.Info("Found NooBaa StorageClass:", storageClassList.Items[0].Name, "with provisioner:", storageClassList.Items[0].Provisioner)
+		return true, nil
+	}
+	
+	klog.Info("No NooBaa StorageClasses found")
+	return false, nil
+}
+
+func checkRGWFlag(ctx context.Context, client ctrlclient.Client, namespace string) (bool, error) {
+	cephObjectStoreExists, err := checkCephObjectStoreExists(ctx, client)
+	if err != nil {
+		return false, fmt.Errorf("failed to check CephObjectStore: %w", err)
+	}
+	if !cephObjectStoreExists {
+		return false, nil
+	}
+
+	storageClassExists, err := checkRGWStorageClassExists(ctx, client, namespace)
+	if err != nil {
+		return false, fmt.Errorf("failed to check RGW StorageClass: %w", err)
+	}
+
+	return storageClassExists, nil
+}
+
+func checkCephObjectStoreExists(ctx context.Context, client ctrlclient.Client) (bool, error) {
+	klog.Info("Checking CephObjectStore existence")
+
+	cephObjectStoreList := &cephv1.CephObjectStoreList{}
+	if err := client.List(ctx, cephObjectStoreList); err != nil {
+		return false, fmt.Errorf("failed to list CephObjectStores: %w", err)
+	}
+
+	if len(cephObjectStoreList.Items) > 0 {
+		klog.Info("Found CephObjectStores, count:", len(cephObjectStoreList.Items))
+		return true, nil
+	}
+
+	klog.Info("No CephObjectStores found")
+	return false, nil
+}
+
+func checkRGWStorageClassExists(ctx context.Context, client ctrlclient.Client, namespace string) (bool, error) {
+	klog.Info("Checking for RGW StorageClass existence")
+	
+	provisioner := fmt.Sprintf("%s.ceph.rook.io/bucket", namespace)
```

Comment:
Provisioner name is hardcoded and doesn't depend on namespace https://github.com/red-hat-storage/ocs-operator/blob/main/controllers/util/csi.go#L4-L8

Reply:
https://github.com/red-hat-storage/ocs-operator/pull/3536#discussion_r2476692314